libs/platform/src/lib/components/table/components/table-p13-dialog/sorting/sorting.component.ts

```diff
@@ -0,0 +1,168 @@
+    readonly isResetAvailable$: Observable<boolean> = this._isResetAvailableSubject$.asObservable();
+
+    /** @hidden */
+    readonly SORT_DIRECTION = SortDirection;
+
+    /** Initial sortBy collection */
+    readonly initialCollectionSort: CollectionSort[];
+
+    /** Sort rules to render */
+    rules: SortRule[] = [];
+
+    /** @hidden */
+    constructor(private dialogRef: DialogRef) {
+        const { columns, collectionSort }: SortDialogData = this.dialogRef.data;
+
+        this.initialCollectionSort = [...collectionSort];
+
+        this.columns = columns || [];
+
+        this._initiateRules();
+    }
+
+    /** Reset changes to the initial state */
+    reset(): void {
+        this._initiateRules();
+        this._isResetAvailableSubject$.next(false);
+    }
+
+    /** Close dialog */
+    cancel(): void {
+        this.dialogRef.close(null);
+    }
+
+    /** Confirm changes and close dialog */
+    confirm(): void {
+        const collectionSort = this._getCollectionSortFromSortRules(this._getUniqueRules(this.rules));
+        const result: SortDialogResultData = { collectionSort: collectionSort };
+        this.dialogRef.close(result);
+    }
+
+    /** @hidden */
+    _removeRule(rule: SortRule): void {
+        this.rules = this.rules.filter((_rule) => _rule !== rule);
+
+        // Keep at least one item in the list
+        if (this.rules.length === 0) {
+            this.rules.push(new SortRule());
+        }
+
+        this._onModelChange();
+    }
+
+    /** @hidden */
+    _addNew(index: number): void {
+        this.rules.splice(index + 1, 0, new SortRule());
+    }
+
+    /** @hidden */
+    _onRuleColumnKeyChange(rule: SortRule, columnKey: string): void {
+        rule.columnKey = columnKey;
+        this._onModelChange();
+    }
+
+    /** @hidden */
+    _onRuleDirectionChange(rule: SortRule, direction: SortDirection): void {
+        rule.direction = direction;
+        this._onModelChange();
+    }
+
+    /** @hidden */
+    _onModelChange(): void {
+        this._isResetAvailableSubject$.next(true);
+    }
+
+    /** @hidden */
+    private _initiateRules(): void {
+        this.rules = this._createRules(this.initialCollectionSort);
+
+        // Keep at least one item in the list
+        if (this.rules.length === 0) {
+            this.rules.push(new SortRule());
+        }
+    }
+
+    /** @hidden */
+    private _createRules(collectionSort: CollectionSort[] = []): SortRule[] {
+        return collectionSort.map(({ field, direction }): SortRule => new SortRule(field, direction));
+    }
+
+    /** @hidden */
+    private _getCollectionSortFromSortRules(rules = this.rules): CollectionSort[] {
+        return rules.filter(this._isRuleValid).map(
+            ({ columnKey, direction }): CollectionSort => ({
+                field: columnKey,
+                direction: direction
+            })
+        );
+    }
+
+    /** @hidden */
+    private _getUniqueRules(rules: SortRule[]): SortRule[] {
```

Comment:
duplicates in two places
libs/platform/src/lib/components/table/components/table-p13-dialog/grouping/grouping.component.ts
libs/platform/src/lib/components/table/components/table-p13-dialog/sorting/sorting.component.ts
Can it be moved to table utils?

Reply:
moved to util